Biology.java

```diff
@@ -0,0 +1,346 @@
+                            // direction of mutation
+                            double theta;
+                            if (Parameters.mut2D) {
+                                theta = Random.nextDouble(0, 2 * Math.PI);
+                            } else {
+                                if (Random.nextBoolean(0.5)) {
+                                    theta = 0;
+                                } else {
+                                    theta = Math.PI;
+                                }
+                            }
+
+                            // size of mutation
+                            double r;
+                            double alpha;
+                            double beta;
+
+                            double meanStep;
+                            double sdStep;
+                            if (this.epitopeSites.contains(nucleotideSiteNumber)) {
+                                // epitope sites
+                                meanStep = Parameters.meanStepEpitope;
+                                sdStep = Parameters.sdStepEpitope;
+                            } else {
+                                // non-epitope sites
+                                meanStep = Parameters.meanStep;
+                                sdStep = Parameters.sdStep;
+                            }
+
+                            r = meanStep;
+                            if (!Parameters.fixedStep) {
+                                alpha = (meanStep * meanStep) / (sdStep * sdStep);
+                                beta = (sdStep * sdStep) / meanStep;
+                                r = Random.nextGamma(alpha, beta);
+                            }
+
+                            // create phenotype
+                            double mutA = r * Math.cos(theta);
+                            double mutB = r * Math.sin(theta);
+
+                            // mutations_i,j
+                            currentSiteMutationMatrix[wildTypeIndex][mutationIndex] = new double[]{mutA, mutB};
+
+                            // mutations_j,i (value is inverse of value at mutations_j,i)
+                            currentSiteMutationMatrix[mutationIndex][wildTypeIndex] = new double[]{-1 * mutA, -1 * mutB};
+
+                            // amino acid mutation notation
+                            // wild type amino acid + site # + mutant amino acid
+                            currentStringOutputCSV += "" + Parameters.AlphabetType.AMINO_ACIDS.getValidCharacters().charAt(wildTypeIndex) +
+                                      nucleotideSiteNumber + Parameters.AlphabetType.AMINO_ACIDS.getValidCharacters().charAt(mutationIndex) +
+                                      "," + r + "," + theta + '\n';
+                        }
+                    }
+                }
+                stringOutputCSV[nucleotideSiteNumber] = currentStringOutputCSV;
```

Comment:
Could you please add a comment that describes what this file is keeping track of?